flang/docs/OpenMPStandardsSupport.md

```diff
@@ -0,0 +1,62 @@
+<!--===- docs/FortranStandardsSupport.md
+
+   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+   See https://llvm.org/LICENSE.txt for license information.
+   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+-->
+
+# Flang OpenMP Standards Support
+
+```{contents}
+---
+local:
+---
+```
+
+This document summarizes OpenMP standards support in Flang. The information is only provided as a guideline. The
+TODOs/Not Yet Implemented messages emitted by the compiler for unimplemented features should be treated as authoritative.
+As of March 2025, standards support is provided upto OpenMP 4.0. We are actively working towards supporting OpenMP 4.5, OpenMP 5.* and OpenMP 6.0.
```

Comment:
I know that it has been requested to have a date in the document, but I would treat the modification date of the file in the commit log as being more accurate than the date in the document that could easily be missed when updating the file (plus, it does not protect us from the document becoming outdated over time).

Reply:
Note that https://clang.llvm.org/docs/OpenMPSupport.html has split OpenMP support into C/C++ & Fortran already. Presumably it makes sense to either fully join those documents, or remove the Fortran side from the clang docs, and just add a reference to this document.

Also typo
```suggestion
As of March 2025, standards support is provided up to OpenMP 4.0. We are actively working towards supporting OpenMP 4.5, OpenMP 5.* and OpenMP 6.0.
```
From the table below, OpenMP 4.0 support is not 100% complete yet, so maybe "up to" gives a wrong impression here; I'd read it as "4.0 support is fully implemented".

Reply:
> Note that https://clang.llvm.org/docs/OpenMPSupport.html has split OpenMP support into C/C++ & Fortran already. Presumably it makes sense to either fully join those documents, or remove the Fortran side from the clang docs, and just add a reference to this document.

Would it be all right to address this separately? Ideally, each Frontend should list its OpenMP support separately. Removing the entry from Clang would require a separate set of reviewers.

Reply:
Rewrote the statement regarding standards support.

Changing the support statement in Clang docs is out of scope for this PR.